src/modules/navigator/navigator_main.cpp

```diff
@@ -1355,6 +1358,40 @@ void Navigator::set_mission_failure_heading_timeout()
 	}
 }
 
+void Navigator::trigger_failsafe(const uint8_t nav_state)
+{
+	if (!_navigator_status.failure || _navigator_status.nav_state != nav_state) {
+		_navigator_status.failure = true;
+		_navigator_status.nav_state = nav_state;
+
+		_navigator_status_updated = true;
+	}
+}
+
+void Navigator::publish_navigator_status()
+{
+	uint8_t current_nav_state = _vstatus.nav_state;
+
+	if (_navigation_mode != nullptr) {
+		current_nav_state = _navigation_mode->getNavigatorStateId();
+	}
+
+	if (_navigator_status.nav_state != current_nav_state) {
+		_navigator_status.nav_state = current_nav_state;
+		_navigator_status.failure = false;
+		_navigator_status_updated = true;
+	}
+
+	if (_navigator_status_updated
+	    || (hrt_elapsed_time(&_last_navigator_status_publication) > 500_ms)) {
```

Comment:
Why do we need to wait half a second here?

Reply:
It's just to at least publish something every 0.5s even if the status didn't change.